target/linux/ramips/mt7621/base-files/etc/uci-defaults/09_fix_crc

```diff
@@ -0,0 +1,18 @@
+. /lib/functions.sh
+
+# don't modify FW data when booting with initramfs image
+fstype="$(/bin/mount | awk '($3 ~ /^\/$/) && ($5 !~ /rootfs/) { print $5 }')"
+[ "$fstype" = "tmpfs" ] && \
+	exit 0
+
+case "$(board_name)" in
+buffalo,wsr-2533dhpls)
+	trx_magic=0x44484C53
+	;;
+*)
+	exit 0
```

Comment:
I do not like the exit in the default branch.
I would prefer if you put the line calculating `kernel_size` and calling `mtd` into a function which gets called for `buffalo,wsr-2533dhpls`.

Reply:
I see, I'll update this.